lib/Backends/CPU/CPUBackend.cpp

```diff
@@ -115,14 +115,13 @@ static void *allocateJITMemory(const IRFunction *F,
 } // end namespace
 
 std::unique_ptr<CompiledFunction>
-CPUBackend::compile(std::unique_ptr<IRFunction> IR) const {
-  AllocationsInfo allocationsInfo;
-  LLVMIRGen irgen(IR.get(), allocationsInfo, "");
+CPUBackend::compileWithIRGen(IRFunction *IR, LLVMIRGen &irgen,
```

Comment:
Instead of having another compile method, could we add a virtual getIRGen method?

The reason I am saying that is because it feels strange to have CPUBackend being non-final will not having any virtual method.

Reply:
@qcolombet I don't have a strong opinion on this. It works for us in both ways. I implemented a solution that I believe implements your suggestion, by adding the virtual `createIRGen` method to `CPUBackend`.